### ISSUE TITLE
Bug: `no-multiple-h1` and `require-alt-text` miss errors after a HTML comment is closed

### ISSUE THREAD

**Issue by ericcornelissen** [eslint/markdown#464]
### Environment

ESLint version: 9.31.0
@eslint/markdown version: 7.0.0
Node version: 22.15.0
npm version: 10.9.2
Operating System: Ubuntu 24.04

### Which language are you using?

gfm

### What did you do?

<details>
<summary>Configuration</summary>

```
import { defineConfig } from "eslint/config";
import markdown from "@eslint/markdown";

export default defineConfig({
  files: ["**/*.md"],
  plugins: { markdown },
  language: "markdown/gfm",
  rules: {
    "markdown/no-multiple-h1": "error",
    "markdown/require-alt-text": "error",
  },
});

```
</details>

```md
# Heading 1

<h1>Reported</h1>
<!-- --> <h1>Not reported</h1>

---

<img src="reported.png" />
<!-- --> <img src="not-reported.png" />
```

### What did you expect to happen?

With [`no-multiple-h1: error`](https://github.com/eslint/markdown/blob/56eb0ee4d1ffd81fe4eb7cec9dda27e35939842f/docs/rules/no-multiple-h1.md) the substring `<h1>Not reported</h1>` should be reported.

With [`require-alt-text: error`](https://github.com/eslint/markdown/blob/56eb0ee4d1ffd81fe4eb7cec9dda27e35939842f/docs/rules/require-alt-text.md) the substring `<img src="not-reported.png" />` should be reported.

### What actually happened?

Neither is reported.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/stackblitz-starters-cttfryej?file=eslint.config.js,markdown.md&view=editor

### Participation

- [ ] I am willing to submit a pull request for this issue.

### Additional comments

For both this is due to the regular subexpression `(?<!<!--[\s\S]*?)<`, which does not end after the closing `-->`.

- https://github.com/eslint/markdown/blob/56eb0ee4d1ffd81fe4eb7cec9dda27e35939842f/src/rules/no-multiple-h1.js#L27
- https://github.com/eslint/markdown/blob/56eb0ee4d1ffd81fe4eb7cec9dda27e35939842f/src/rules/require-alt-text.js#L27
